test/unit/org/openstreetmap/josm/gui/dialogs/properties/PropertiesDialogFilterActionTest.java

```diff
@@ -0,0 +1,197 @@
+        } catch (Exception e) {
+            fail("Should not throw", e);
+        }
+    }
+
+    @Test
+    void filterActionShowOnlyTest() {
+        DataSet ds = new DataSet();
+        OsmDataLayer layer = new OsmDataLayer(ds, "", null);
+        MainApplication.getLayerManager().addLayer(layer);
+        MapFrame mf = MainApplication.getMap();
+        PropertiesDialog propertiesDialog = mf.propertiesDialog;
+        FilterDialog filterDialog = mf.filterDialog;
+        FilterTableModel filterModel = filterDialog.getFilterModel();
+
+        Node n = new Node(new EastNorth(0, 0));
+        n.put("natural", "tree");
+        n.put("leaf_type", "broadleaved");
+        ds.addPrimitive(n);
+
+        Node n2 = new Node(new EastNorth(0, 0));
+        n2.put("natural", "tree");
+        n2.put("leaf_type", "needleleaved");
+        ds.addPrimitive(n2);
+
+        Node n3 = new Node(new EastNorth(0, 0));
+        n3.put("natural", "tree");
+        ds.addPrimitive(n3);
+
+        try {
+            Field filterActionShowOnlyField = PropertiesDialog.class.getDeclaredField("filterActionShowOnly");
+            filterActionShowOnlyField.setAccessible(true);
+            FilterAction filterActionShowOnly = (PropertiesDialog.FilterAction) filterActionShowOnlyField.get(propertiesDialog);
+
+            Field tagTableField = PropertiesDialog.class.getDeclaredField("tagTable");
+            tagTableField.setAccessible(true);
+            JTable tagTable = (JTable) tagTableField.get(propertiesDialog);
+
+            Field tagDataField = PropertiesDialog.class.getDeclaredField("tagData");
+            tagDataField.setAccessible(true);
+            ReadOnlyTableModel tagData = (ReadOnlyTableModel) tagDataField.get(propertiesDialog);
+
+            assertEquals(0, tagData.getRowCount());
+            ds.setSelected(n, n2);
+            assertEquals(2, tagData.getRowCount());
+
+            tagTable.selectAll();
+
+            assertEquals(0, filterModel.getFilters().size());
+            filterActionShowOnly.actionPerformed(null);
+            assertEquals(1, filterModel.getFilters().size());
+
+            SearchSetting expectedSearchSetting = new SearchSetting();
+            expectedSearchSetting.text = "((\"leaf_type\"=\"broadleaved\") OR (\"leaf_type\"=\"needleleaved\")) AND (\"natural\"=\"tree\")";
+            expectedSearchSetting.caseSensitive = true;
+            Filter expectedFilter = new Filter(expectedSearchSetting);
+            expectedFilter.inverted = true;
+
+            Filter createdFilter = filterModel.getFilters().get(0);
+
+            assertEquals(expectedFilter, createdFilter);
+            assertEquals(0, createdFilter.compareTo(expectedFilter));
+            assertEquals(true, createdFilter.enable);
+
+            assertEquals(false, n.isDisabled());
+            assertEquals(false, n2.isDisabled());
+            assertEquals(true, n3.isDisabled());
+        } catch (Exception e) {
+            fail("Should not throw", e);
```

Comment:
I don't know if you are aware of `assertDoesNotThrow` or not, but you can use that instead of catching the exception.